### PR TITLE
rfc: dx improvements

### PR DESCRIPTION
**Pull request by PhearZero**
# ℹ️ Overview

This pr consolidates a few DX upgrades for the project. While we have a gap in xgov sessions it would be good to solve what we can. 

_These features do not take a large amount of time to implement but it will require coordination with any features in development._

## 📝TODO
- [x] converts to workspace
- [x] merges `shared/types` and includes "Type Guards" for `RoundInfo.voteType`
- [x] moves  `shared/*` (VotingRoundContract, IPFSGateway, csvSigner)
- [x] removes `dapp` 
- [x] uses `src` alias as `@`
- [x] preview deployments (Vercel, Chromatic, etc)
- [ ] Resolve feature conflicts or reapply changes
- [ ] Optimize CI/CD for workspaces

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -14,19 +14,19 @@ permissions:
...
 
 jobs:
-  build-dapp:
-    name: Dapp
-    uses: ./.github/workflows/node-ci.yml
-    with:
-      working-directory: ./src/dapp
-      run-build: true
-      audit-script: npm run audit
+#  build-dapp:
```
Comment (ehanoc) [inline .github/workflows/pr.yml:17]: If commented, maybe it can be removed?

File: infrastructure/bin/infrastructure.ts [hunk 2]
```diff
@@ -93,12 +93,12 @@ const responseHeaders: ResponseHeadersPolicyProps = {
   removeHeaders: ['server'],
 }
 
-const app = deployer.deploy(StaticWebsiteStack, 'web', {
-  websiteFolder: process.env.WEBSITE_BUILD_PATH ?? path.join(__dirname, '..', '..', 'src', 'dapp', 'dist'),
-  websiteNpmBuildCommand: 'build-dapp',
-  customDomain: dns?.getDefaultCustomDomainProps(deployer.defaultRegion, appDomainName),
-  responseHeaders: responseHeaders,
-})
+// const app = deployer.deploy(StaticWebsiteStack, 'web', {
```
Comment (ehanoc) [inline infrastructure/bin/infrastructure.ts:96]: comments? Maybe can be removed?

File: src/xgov-dapp/src/shared/types.ts [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
-  voteInformationUrl?: string
+  voteDescription?: string | undefined
   start: string
   end: string
   proposalFile: string
   snapshotFile?: string | undefined
+  minimumVotes?: number | undefined
+  voteInformationUrl?: string
 }
+// export type RoundInfo = {
```
Comment (ehanoc) [inline src/xgov-dapp/src/shared/types.ts:33]: comments :)

**Review by ehanoc — COMMENTED** [review]
removals and improvements. I like it :+1:

File: .github/workflows/smart-contract-ci.yml [hunk 4]
```diff
@@ -51,8 +51,8 @@ jobs:
         with:
           node-version: ${{ inputs.node-version }}
           cache: "npm"
-          cache-dependency-path: ${{ inputs.working-directory }}/smart_contracts/package-lock.json
-      
+          cache-dependency-path: ./package-lock.json
```
Comment (robdmoore) [inline .github/workflows/smart-contract-ci.yml:54]: Will this work? There is still a package-lock in the sub-directory and I'm not sure if GitHub cache works off of a parent package.json with workspaces.

Comment (PhearZero) [reply .github/workflows/smart-contract-ci.yml:54]: Shouldn't be any package-lock's outside of the root directory and infrastructure(not in workspace). Pretty sure it's supported, worst case scenario we can implement our own cache strategy. Still need to refactor the CI/CD pipeline to be in-line with workspaces. (running install once and recursive running of scripts). 

This is an experimental branch of changes found during discovery and may end up being dropped and reintroduced (depending on backlog pressure). Very much a WIP

Comment (robdmoore) [reply .github/workflows/smart-contract-ci.yml:54]: Sure, makes sense if you are going to house the package-lock at that position. Worth pointing out though that you'll want to check that the output of npm run build for each sub-component works with the cdk code. If you have trouble hit me up.

**Pull request closed without merging**